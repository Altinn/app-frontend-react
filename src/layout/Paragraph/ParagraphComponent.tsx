import React from 'react';

import { Grid, Typography } from '@material-ui/core';

import { HelpTextContainer } from 'src/components/form/HelpTextContainer';
import { Lang } from 'src/features/language/Lang';
import { useLanguage } from 'src/features/language/useLanguage';
import type { PropsFromGenericComponent } from 'src/layout';

export type IParagraphProps = PropsFromGenericComponent<'Paragraph'>;

export function ParagraphComponent({ node }: IParagraphProps) {
  const { id, textResourceBindings } = node.item;
  const { langAsString } = useLanguage();
  return (
    <Grid
      container={true}
      direction='row'
      alignItems='center'
      spacing={1}
    >
      <Grid item={true}>
        <Typography
          component={'div'}
          id={id}
          data-testid={`paragraph-component-${id}`}
        >
          <Lang id={textResourceBindings?.title} />
        </Typography>
      </Grid>
      {textResourceBindings?.help && (
        <Grid item={true}>
          <HelpTextContainer
            helpText={<Lang id={textResourceBindings.help} />}
            title={langAsString(textResourceBindings?.title)}
          />
        </Grid>
      )}
    </Grid>
  );
}
