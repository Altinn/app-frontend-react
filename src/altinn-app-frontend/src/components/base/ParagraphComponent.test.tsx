import * as React from 'react';
import { render, screen } from '@testing-library/react';
import type { ITextResourceBindings } from 'src/features/form/layout';
import type { IComponentProps } from 'src/components';

import { ParagraphComponent } from './ParagraphComponent';

describe('ParagraphComponent', () => {
  const mockId = 'mock-id';
  const mockText = 'Here goes a paragraph';
  const mockGetTextResource = (key: string) => key;
  const mockLanguage: any = {};
  const mockTextResourceBindings: ITextResourceBindings = {
    tile: mockText,
  };

  it('should match snapshot', () => {
    const { container } = render(
      <ParagraphComponent
        id={mockId}
        text={mockText}
        language={mockLanguage}
        getTextResource={mockGetTextResource}
        textResourceBindings={mockTextResourceBindings}
        {...({} as IComponentProps)}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have correct text', () => {
    render(
      <ParagraphComponent
        id={mockId}
        text={'some other text'}
        language={mockLanguage}
        getTextResource={mockGetTextResource}
        textResourceBindings={mockTextResourceBindings}
        {...({} as IComponentProps)}
      />,
    );
    expect(screen.getByText('some other text')).toBeInTheDocument();
  });

  it('should render help text if help text is supplied', () => {
    render(
      <ParagraphComponent
        id={mockId}
        text={mockText}
        language={mockLanguage}
        getTextResource={mockGetTextResource}
        textResourceBindings={{ help: 'this is the help text' }}
        {...({} as IComponentProps)}
      />,
    );
    expect(
      screen.getByRole('button', {
        name: /popover\.popover_button_helptext/i,
      }),
    ).toBeInTheDocument();
  });
});
