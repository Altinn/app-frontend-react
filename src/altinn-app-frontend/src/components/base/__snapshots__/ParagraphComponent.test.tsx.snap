// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParagraphComponent should match snapshot 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
>
  <div
    class="MuiGrid-root MuiGrid-item"
  >
    <p
      class="MuiTypography-root makeStyles-spacing-1 makeStyles-typography-2 MuiTypography-body1"
      id="mock-id"
    >
      Here goes a paragraph
    </p>
  </div>
</div>
`;
